src/main/json2xlsx.js

```diff
@@ -22,31 +23,23 @@ const run = async (args) => {
   const builder = new ccdUtils.SpreadsheetBuilder(sourceXlsx);
   await builder.loadAsync();
 
-  const excludedFilenamePatterns = args.excludedFilenamePatterns ? stringUtils.split(args.excludedFilenamePatterns) : [];
-  const files = fileUtils.listFilesInDirectory(args.sheetsDir, excludedFilenamePatterns);
-
-  for (const file of files) {
-    const readSheetData = async (file) => {
-      const readJsonFile = (relativeFilePath) => {
-        return fileUtils.readJson(path.join(args.sheetsDir, relativeFilePath), Substitutor.injectEnvironmentVariables);
-      };
-
-      if (file.isDirectory()) {
-        const jsonFragments = await Promise.all(
-          fileUtils.listFilesInDirectory(path.join(args.sheetsDir, file.name), excludedFilenamePatterns)
-            .map(fragmentFile => readJsonFile(`${file.name}/${fragmentFile.name}`))
-        );
-        return jsonFragments.flat();
-      } else {
-        return await readJsonFile(file.name);
-      }
-    };
-
-    console.log(`  importing sheet data from ${file.name} ${file.isDirectory() ? 'directory' : 'file'}`);
-    const json = await readSheetData(file);
-    ccdUtils.JsonHelper.convertPropertyValueStringToDate('LiveFrom', json);
-    ccdUtils.JsonHelper.convertPropertyValueStringToDate('LiveTo', json);
-    builder.updateSheetDataJson(path.basename(file.name, '.json'), json);
+  let excludedFilenamePatterns = args.excludedFilenamePatterns
+    ? stringUtils.split(args.excludedFilenamePatterns) : [];
+
+  let fileMap = await fileUtils.getJsonFiles(args.sheetsDir,
+    excludedFilenamePatterns);
+
+  for (const file in fileMap) {
+    if (Object.prototype.hasOwnProperty.call(fileMap, file)) {
```

Comment:
What is that `if` for? Is there any real use case which makes that `if` necessary? Any properties that comes from prototype?

Reply:
It was to remove a warning about possible iteration over unexpected members although considering that this is just an object that I construct would it be necessary (still learning js)